src/ucp/rndv/rndv_rkey_ptr.c

```diff
@@ -268,6 +276,13 @@ ucp_proto_rndv_rkey_ptr_mtype_probe(const ucp_proto_init_params_t *init_params)
         return;
     }
 
+    if (ucp_proto_rndv_init_params_incl_prev_stages(init_params)) {
+        status = ucp_proto_rndv_predict_prev_stages(init_params, &caps);
```

Comment:
can you move check for `ucp_proto_rndv_init_params_incl_prev_stages` to `ucp_proto_rndv_predict_prev_stages`? To avoid a bit of code duplication

Reply:
Done.